sympy/combinatorics/fp_groups.py

```diff
@@ -513,6 +513,16 @@ def elements(self):
         P, T = self._to_perm_group()
         return T.invert(P._elements)
 
+    def is_cyclic(self):
+        """
+        Return ``True`` if group is Cyclic.
+
+        """
+        if len(self.generators) == 0 or len(self.generators) == 1:
+            return True
+        P, T = self._to_perm_group()
```

Comment:
`_to_perm_group` will raise `NotImplementedError` if the group is infinite. That can be raised also here but the error message should probably be changed. Maybe something like this:

    try:
        P, _ = self._to_perm_group()
    except NotImplementedError:
        raise NotImplementedError("suitable message")